pkg/endpoint/policy.go

```diff
@@ -764,6 +747,12 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	// Mark the endpoint to be running the policy revision it was
 	// compiled for
 	e.bumpPolicyRevisionLocked(revision)
+
+	// Set to Ready if the endpoint has not already transitioned further
+	if e.GetStateLocked() == StateRegenerating {
```

Comment:
Why isn't this logic inside the `BuilderSetStateLocked`?

Reply:
This clearly documents that the state may already have transitioned and that it's fine if that happened. This was absolutely not clear to me before if this is hidden in a function that is supposed to *set* the state.